### ISSUE TITLE
Clone option in the list view is missing on Maps for non admin users

### ISSUE THREAD

**Issue by marthamareal** [GeoNode/geonode-mapstore-client#1308]
#### Expected behavior
Users should be able to clone Maps from the list view.
#### Actual behavior
Only admins can see the clone option on maps in the list view

**Comment by DavidQuartz** [comment]
We have 2 requirements for the clone action are the `is_copyable` prop and `download_resourcebase` perm. But somehow for maps, only admins have the `download_resourcebase` perm on them so [this check](https://github.com/GeoNode/geonode-mapstore-client/blob/master/geonode_mapstore_client/client/js/components/CardGrid/CardGrid.jsx#L96) doesn't pass for normal users.

It seems the permissions for maps need to be checked for normal users on the server side, if the behaviour is not expected @marthamareal @giohappy

**Comment by giohappy** [comment]
@mattiagiupponi can you take a look at this?

**Comment by mattiagiupponi** [comment]
@giohappy 
The permission is missing because `map` as a subtype is not considered as [DOWNLOADABLE_RESOURCES](https://github.com/GeoNode/geonode/blob/477c4bbc6c0d17ba87a90200751194cb2867126e/geonode/security/permissions.py#L68-L71) saying that:
- [Here the permissions ](https://github.com/GeoNode/geonode/blob/477c4bbc6c0d17ba87a90200751194cb2867126e/geonode/resource/manager.py#L701-L702)are assigned for the resource:
```python
elif AdvancedSecurityWorkflowManager.assignable_perm_condition(perm, _resource_type):
    _safe_assign_perm(perm, anonymous_group, _resource.get_self_resource())
```
Internally the `AdvancedSecurityWorkflowManager` evaluate if the resource can have the "download_resourcebase" permissions if is in the `DOWNLOADABLE_RESOURCES` list
```python
_assignable_perm_policy_condition = (perm in DOWNLOAD_PERMISSIONS and resource_type in DOWNLOADABLE_RESOURCES) or \
            (perm in DATASET_EDIT_DATA_PERMISSIONS and resource_type in DATA_EDITABLE_RESOURCES_SUBTYPES) or \
            (perm not in (DOWNLOAD_PERMISSIONS + DATASET_EDIT_DATA_PERMISSIONS))
```
The map doesn't pass mainly because of the first if since `map` is not in the [DOWNLOADABLE_RESOURCES](https://github.com/GeoNode/geonode/blob/477c4bbc6c0d17ba87a90200751194cb2867126e/geonode/security/permissions.py#L68-L71) so that the `download_resourcebase` permission is not assigned.

NOTE: we can add the map in the `DOWNLOAD_PERMISSIONS` but I guess this will enable other functionalities in the front end since the map cannot be downloaded. A suggestion could be to let a resource be cloned if the `is_copyable` is True and if the user has the `add_resourcebase` permission

**Comment by DavidQuartz** [comment]
In that case, for admins, are all possible perms (which includes download_resourcebase) automatically returned for maps? 
@mattiagiupponi

**Comment by mattiagiupponi** [comment]
> In that case, for admins, are all possible perms (which includes download_resourcebase) automatically returned for maps? @mattiagiupponi

Yes, the admin will have all the permissions because it does not pass through that code

> I suspect that specific permission must be created to say if the user has permission to clone, which will be calculated differently based on the resource type. Might it be copy_resourcebase?

Can't it be easier to rely only on the `is_copyable` property? If we make the property the central point of reliability, is also easier to maintain. Will be the property to calculate when the given resource is copyable or not

**Comment by giohappy** [comment]
>@giohappy Can't it be easier to rely only on the is_copyable property? If we make the property the central point of reliability, is also easier to maintain (and test). Will be the property to calculate when the given resource is copyable or not

`is_copyable` is a property of the resource, not permission. It only says that granted you have permission to do it, the resource can be copied. For example, it is `false` for remote datasets.

**Comment by DavidQuartz** [comment]
> By the way @DavidQuartz I see that the "Save as..." button is available inside a map. How is the visibility of this button calculated?

@giohappy the Save As plugin is allowed on all maps in map viewer. The permissions are only checked for datasets and documents as [this issue](https://github.com/geonode/geonode-mapstore-client/issues/1177) required

**Comment by giohappy** [comment]
as agreed with @allyoucanmap for the moment we will let the client decide if `download_resourcebase` must be taken into account or not depending on the resource type.

 - it will be implemented on master according to the new catalog code
 - it will be fixed in 4.0.x
